### DISCUSSION TITLE
Add support for language packs

### DISCUSSION THREAD

**Discussion by Kodalinq** [solidtime-io/solidtime#256] [Feature requests]
Hello, SolidTime team!

Thank you for your work on SolidTime — it’s truly a useful and user-friendly application. I understand that such a request might spark some debate, but I would like to share an idea that could make the application even more user-friendly.

It would be great to see support for language packs, allowing users to choose their preferred language. For example, adding Russian would make the application accessible to a large number of Russian-speaking users.

This proposal includes:

The ability to use translation files for core UI elements (e.g., notifications, buttons, errors).
Documentation or guidance on adding new languages.
(Optional) Adding Russian as a default translation.
If language pack support is already in development, I’d love to hear about it! 

Thank you for your attention, and please keep up the great work on this excellent product!

**Comment by Onatcer** [comment]
Thanks for the suggestion. We plan to add multi-language support in the future, but we'll have to adapt to the frontend before doing that, so it might take a while before we get to it. We are also very much looking forward to work together with the international open-source community on this.

If there are any suggestions for tooling to manage the translations, feel free to let us know below, and also upvote the feature request so we can see how many people are interested in this.

**Comment by Kodalinq** [comment]
Thank you for your response! I’m very glad to hear that multi-language support is already in your plans, and I understand the need to adapt the frontend before implementing this feature.

Here are a few suggestions for managing translations:

Translation Management Tools:

Crowdin: A powerful tool for localization with community collaboration.
Weblate: Open-source software that integrates well with repositories.
POEditor: A simple and efficient solution for managing translations in various formats.
Lokalise: A robust platform for teams working on multilingual applications.

I’ll definitely upvote the feature request and share it with others who might be interested. 

Looking forward to seeing this feature in SolidTime!

**Comment by murray3k** [comment]
Multi-language support would be awesome! Then I could get all my German colleges to using SolidTime.
But I also understand, that the process of internationalization is a daunting task and a laborious task.

My opinions for tooling are:

For the frontend there two obvious choices being:
- Format.js (https://formatjs.github.io/docs/vue-intl/#tooling)
- i18next (https://i18next.github.io/i18next-vue/)

Format.js is more standards orientated and has less features as i18next, but both are fine libraries. I prefer i18next for the nice plugins like a browser-languagedetector and others.

Either way that means all translations will be stored in json files, which would be also fine for laravel, but the formats are not really compatible. So your project will have two separate translations for backend & frontend. Should be fine.
Also both json formats are supported by the major community translation tools. 

There I would prefer Weblate as a true open-source solution and they offer free hosting for open source projects. Crowdin is also fine and does offer a free tier for open source project too, but their service seems to collect much more userdata for their AI stuff.

Looking forward for this feature and would gladly provide a German translation. Let me know if you have any questions.

**Comment by lexcorp** [comment]
I'm ready to help to translate to spanish...

**Comment by m-a-mohsen** [comment]
If you are planning to include RTL languages such as "Arabic, Hebrew or Farsi", I think it is safer to go with Nuxt UI components [(https://ui.nuxt.com/docs/getting-started/installation/vue)](https://ui.nuxt.com/docs/getting-started/installation/vue) rather than Shadcn [https://github.com/shadcn-ui/ui/issues/5776](https://github.com/shadcn-ui/ui/issues/5776)

**Comment by m-a-mohsen** [comment]
Now shadcn@3.8.0 supports RTL since Jan 2026

**Comment by pipozzz** [comment]
I can help you out with Slovak translation

**Comment by iamspido** [comment]
Thanks for your great work. Solidtime is great!
Is there an estimated timeline for when multi-language support will be implemented (e.g., month, quarter, or year)?
Also, do you have any kind of roadmap available for Solidtime?

**Comment by pipozzz** [comment]
any update?

**Comment by Muecke1978** [comment]
Any update?
